src/java.base/share/classes/jdk/internal/classfile/impl/AbstractAttributeMapper.java

```diff
@@ -62,13 +62,14 @@ public final String name() {
     }
 
     @Override
-    public final void writeAttribute(BufWriter buf, T attr) {
+    public final void writeAttribute(BufWriter writer, T attr) {
+        BufWriterImpl buf = (BufWriterImpl) writer;
         buf.writeIndex(buf.constantPool().utf8Entry(name));
         buf.writeInt(0);
```

Comment:
Since this is the int we're patching later it might be reasonable to have a buf.skip(int) method which just increases buf.offset. Applicable at least here and in `UnboundAttribute::writeTo`